Lib9c/Model/State/HammerPointState.cs

```diff
@@ -30,9 +32,9 @@ public IValue Serialize()
                 .Add(HammerPoint.Serialize());
         }
 
-        public void AddHammerPoint(int point)
+        public void AddHammerPoint(int point, CrystalHammerPointSheet sheet)
         {
-            HammerPoint += point;
+            HammerPoint = Math.Min(HammerPoint + point, sheet[RecipeId].MaxPoint);
```

Comment:
It will be better to do a `null` check for the `sheet` argument.

Reply:
OK I will add that!